x/stakeibc/keeper/ibc_test.go

```diff
@@ -0,0 +1,291 @@
+	icaAccountTypes := []types.ICAAccountType{
+		types.ICAAccountType_DELEGATION,
+		types.ICAAccountType_WITHDRAWAL,
+		types.ICAAccountType_REDEMPTION,
+		types.ICAAccountType_FEE,
+		types.ICAAccountType_COMMUNITY_POOL_DEPOSIT,
+		types.ICAAccountType_COMMUNITY_POOL_RETURN,
+
+		types.ICAAccountType_CONVERTER_TRADE, // not on the host zone
+		-1,                                   // indicates test case for non-ICA port
+	}
+
+	for _, accountType := range icaAccountTypes {
+		// Reset the host zone for each test and wipe all addresses
+		s.App.StakeibcKeeper.SetHostZone(s.Ctx, types.HostZone{
+			ChainId:                           HostChainId,
+			DepositAddress:                    DepositAddress,
+			CommunityPoolStakeHoldingAddress:  CommunityPoolStakeHoldingAddress,
+			CommunityPoolRedeemHoldingAddress: CommunityPoolRedeemHoldingAddress,
+		})
+
+		// Determine the port Id from the account type
+		// If the portId is -1, pass a non-ica port
+		portId := "not-ica-port"
+		if accountType != -1 {
+			owner := types.FormatICAAccountOwner(HostChainId, accountType)
+			portId, _ = icatypes.NewControllerPortID(owner)
+		}
+
+		// Call StoreHostZoneIcaAddress with the portId
+		// use the account name as the address to make the matching easier
+		address := accountType.String()
+		err := s.App.StakeibcKeeper.StoreHostZoneIcaAddress(s.Ctx, HostChainId, portId, address)
+		s.Require().NoError(err, "no error expected when calling store host zone ICA for %s", accountType.String())
+
+		// Check if the updated addresses matches expectations
+		s.checkHostZoneAddressStored(accountType)
+
+		// Check that the relevant accounts are white listed from the rate limiter
+		s.checkAddressesWhitelisted(accountType)
+	}
+}
+
+// ------------------------------------------
+//      	StoreTradeRouteIcaAddress
+// ------------------------------------------
+
+// Helper function to check that a single ICA address was stored on the trade route
+// The address stored will match the string of the ICA account type
+func (s *KeeperTestSuite) checkTradeRouteAddressStored(accountType types.ICAAccountType) {
+	// Determine the expected ICA addresses based on whether the account in question
+	// is registered in this test case
+	unwindAddress := ""
+	if accountType == types.ICAAccountType_CONVERTER_UNWIND {
+		unwindAddress = types.ICAAccountType_CONVERTER_UNWIND.String()
+	}
+	tradeAddress := ""
+	if accountType == types.ICAAccountType_CONVERTER_TRADE {
+		tradeAddress = types.ICAAccountType_CONVERTER_TRADE.String()
+	}
+
+	// Confirm the expected addresses with the host zone
+	tradeRoute, found := s.App.StakeibcKeeper.GetTradeRoute(s.Ctx, RewardDenom, HostDenom)
+	s.Require().True(found, "trade route should have been found")
+
+	s.Require().Equal(unwindAddress, tradeRoute.RewardAccount.Address, "unwind address")
+	s.Require().Equal(tradeAddress, tradeRoute.TradeAccount.Address, "trade address")
+}
```

Comment:
can you elaborate on what the change would be?

Reply:
I was thinking before that we should include a setup test around withdrawal address because it goes through both the HostZone and it needs to show up on the TradeRoute -- but I don't think that it matters now since we assume the HostZone is always initialized first before the TradeRoute and we get the withdrawal address from the HostZone.  I no longer think this adds anything beyond the delegation address setup